scripts/expX2.py

```diff
@@ -0,0 +1,763 @@
+# Warning: if this value is under-estimated, it will happen the cluster will kill jobs
+# that are not finished withing the specified time.
+MINUTES_PER_RUN = int(sys.argv[7])
+
+# How many scripts M we want the N jobs to be divided into.
+# Note: on cluster we can at most submit 400 scripts.
+# Also not that in the same .sh script there can be experiments only for a single SUT.
+NJOBS = int(sys.argv[8])
+
+# Specify a string to filter CONFIGS to be included
+# None or `all` represents all CONFIGS should be included
+# Default is None
+CONFIGFILTER = None
+if len(sys.argv) > 9:
+    CONFIGFILTER = str(sys.argv[9])
+
+# Specify a string to filter suts to be included based on their names
+# None or `all` represents all Suts should be included
+# Default is None
+SUTFILTER = None
+if len(sys.argv) > 10:
+    SUTFILTER = str(sys.argv[10])
+
+# input parameter validation
+if MIN_SEED > MAX_SEED:
+    print("ERROR: min seed is greater than max seed")
+    exit(1)
+
+if not os.path.isdir(BASE_DIR):
+    print("creating folder: " + BASE_DIR)
+    os.makedirs(BASE_DIR)
+else:
+    print("ERROR: target folder already exists")
+    exit(1)
+
+
+JDK_8 = "JDK_8"
+JDK_11 = "JDK_11"
+JS = "JS"
+DOTNET_3 = "DOTNET_3"
+
+class Sut:
+    def __init__(self, name, timeWeight, platform):
+        self.name = name
+        # the higher value, the more time it will need compared to the other SUTS
+        self.timeWeight = timeWeight
+        # Java? JS? NodeJS
+        self.platform = platform
+
+
+
+# To ge the SUTs, you need in EMB to run the script "scripts/dist.py" to
+# generate a dist.zip file that you can upload on cluster.
+# Note: the values after the SUT names is multiplicative factor for how long
+# experiments should be run.
+# Depending on what experiments you are running, might want to de-select some
+# of the SUTs (eg, by commenting them out)
+
+SUTS = [
+        # REST
+        Sut("features-service", 1, JDK_8),
+        Sut("scout-api", 2, JDK_8),
+        Sut("proxyprint", 2, JDK_8),
+        Sut("rest-ncs", 2, JDK_8),
+        Sut("rest-scs", 1, JDK_8),
+        Sut("rest-news", 1, JDK_8),
+        Sut("catwatch", 1, JDK_8),
+        Sut("restcountries", 2, JDK_8),
+        Sut("languagetool", 3, JDK_8),
+        # GRAPHQL
+        Sut("petclinic", 1, JDK_8),
+        Sut("patio-api", 1, JDK_11),
+        Sut("timbuctoo", 1, JDK_11),
+        Sut("graphql-ncs", 1, JDK_8),
+        Sut("graphql-scs", 1, JDK_8),
+        # Sut("ind0", 1, JDK_8),
+        # Sut("ocvn-rest", 1, JDK_8),
+        # Sut("ncs-js", 1, JS),
+        # Sut("scs-js", 1, JS)
+        # .Net
+        Sut("cs-rest-ncs",1,DOTNET_3),
+        Sut("cs-rest-scs",1,DOTNET_3),
+        Sut("sampleproject",1,DOTNET_3),
+        Sut("menu-api",1,DOTNET_3)
+]
+
+if SUTFILTER is not None and SUTFILTER.lower() != "all":
+    filteredsut = list(filter(lambda x: x.name.lower() in (f.lower() for f in SUTFILTER.split(",")), SUTS))
+    if len(filteredsut) > 0:
```

Comment:
should give an error if user put a SUT name that does not exist

Reply:
rather than issuing error if none exists, should issue error if any does not exist.